### PR TITLE
Add option to initialize repository with labels

### LINKED ISSUE #6055: Automatically add labels to repositories on creation
- Gitea version (or commit ref): 7f38e2d0dedfc3e04994272196cda53e4ad20354
- Git version: git version 2.17.2 (Apple Git-113)
- Operating system: MacOS
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [ ] MySQL
  - [ ] MSSQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [x] Yes (provide example URL)
  - [ ] No
  - [ ] Not relevant
- Log gist:

## Description

I expect to be able to create a repo, file issues and add labels immediately. 
Currently, I have to manually enable a label set individually created for every 
single repository.

### PR DESCRIPTION
**Pull request by jolheiser**
Resolves #6055
Adds a dropdown for users to select a label set when initializing a repository.  
  
issue labels repo creation

### CODE REVIEW

**Comment by jolheiser** [comment]
This would address #6055

**Comment by jolheiser** [comment]
Resolves #6055

File: models/repo.go [hunk 1]
```diff
@@ -1177,6 +1178,30 @@ func prepareRepoCommit(e Engine, repo *Repository, tmpDir, repoPath string, opts
 		}
 	}
 
+	// Issue Labels
```
Comment (lafriks) [inline models/repo.go]: This should be moved out to separate function as this is not related to initializing `git` structure and function call should be added to https://github.com/go-gitea/gitea/blob/cf8eb1ab9ce3f882b399601292f456023a6332dc/models/repo.go#L1408

**Review by lafriks — CHANGES_REQUESTED** [review]
imho label sets should be created always if selected even if you do not select to initialize git repository (except if it is mirror of course)

**Comment by lafriks** [comment]
sorry for such late review

**Comment by lafriks** [comment]
Also it should be moved up before `.gitignore` selection just below description.

**Comment by lafriks** [comment]
Please update screenshot. Also this option should probably needs to be added also to API for creating new repo

**Comment by jolheiser** [comment]
@lafriks Here's an updated image. Was this what you had in mind?

Issue Labels Repo Creation New

**Comment by lafriks** [comment]
Yes, so we could add this to release notes

File: models/repo.go (hunk 1 above)
Comment (jolheiser) [reply models/repo.go]: Done

File: models/issue_label.go [hunk 2]
```diff
@@ -127,6 +127,33 @@ func (label *Label) ForegroundColor() template.CSS {
 	return template.CSS("#000")
 }
 
+func initalizeLabels(e Engine, repoID int64, labelTemplate string) error {
```
Comment (lafriks) [inline models/issue_label.go:130]: Could you reuse this function here: https://github.com/go-gitea/gitea/blob/f9ec2f89f2265bc1371a6c62359de9816534fa6b/routers/repo/issue_label.go#L36

File: models/issue_label.go [hunk 3]
```diff
@@ -127,6 +127,33 @@ func (label *Label) ForegroundColor() template.CSS {
...
+		labels[i] = &Label{
+			RepoID:      repoID,
+			Name:        list[i][0],
+			Description: list[i][2],
+			Color:       list[i][1],
+		}
+	}
+	for _, label := range labels {
+		if err = newLabel(e, label); err != nil {
+			return fmt.Errorf("newLabel: %v", err)
```
Comment (lafriks) [inline models/issue_label.go]:
```suggestion
			return err
```

**Review by lafriks — CHANGES_REQUESTED** [review]

File: models/issue_label.go (hunk 2 above)
Comment (jolheiser) [reply models/issue_label.go:130]: I actually did the opposite. The router now uses _this_ function. (More specifically it uses the public version `InitializeLabels`)

Comment (lafriks) [reply models/issue_label.go:130]: Oh, sorry did not notice you have already done this :+1:

**Review by lafriks — APPROVED** [review]

**Comment by lafriks** [comment]
Nice work, thanks ;)

File: models/issue_label.go [hunk 4]
```diff
@@ -127,6 +127,33 @@ func (label *Label) ForegroundColor() template.CSS {
...
+	}
+	for _, label := range labels {
+		if err = newLabel(e, label); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func InitalizeLabels(repoID int64, labelTemplate string) error {
```
Comment (guillep2k) [inline models/issue_label.go:154]:
```suggestion
// InitalizeLabels adds a label set to a repository using a template
func InitalizeLabels(repoID int64, labelTemplate string) error {
```

Comment (guillep2k) [reply models/issue_label.go:154]: Required for build to succeed

**Review by guillep2k — CHANGES_REQUESTED** [review]

File: models/issue_label.go (hunk 4 above)
Comment (jolheiser) [reply models/issue_label.go:154]: Ah, oversight. Thanks!

**Review by guillep2k — APPROVED** [review]
